src/clipboard.c

```diff
@@ -2228,4 +2328,545 @@ adjust_clip_reg(int *rp)
     }
 }
 
+#if defined(FEAT_WAYLAND_CLIPBOARD) || defined(PROTO)
+
+/*
+ * Read data from a file descriptor and write it to the given clipboard.
+ */
+    static void
+clip_wl_receive_data(Clipboard_T *cbd, const char *mime_type, int fd)
+{
+    char_u *start, *buf, *tmp, *final, *enc;
+    int	motion_type = MAUTO;
+    ssize_t r = 0;
+    size_t total = 0, max_total = 4096; // Initial buffer size, 4096 bytes seems
+					// reasonable.
+#ifndef HAVE_SELECT
+    struct pollfd pfd = {
+	.fd = fd,
+	.events = POLLIN
+    };
```

Comment:
we don't use designated initializer in Vim. Not sure if all systems support those.